generatorreceiver/internal/generator/trace_generator.go

```diff
@@ -95,10 +78,8 @@ func (g *TraceGenerator) createSpanForServiceRouteCall(traces *pdata.Traces, ser
 	resource.Attributes().InsertString(string(semconv.ServiceNameKey), serviceTier.ServiceName)
 
 	resourceAttributeSet := serviceTier.GetResourceAttributeSet()
-	if resourceAttributeSet != nil {
```

Comment:
We don't need this check because the zero value for a `Pickable` / `ResourceAttributeSet` is not `nil`.